cli: add rad fork command

FintanH opened the pull request:
Add a `rad fork` command to create a forked namespace, using the local peer, for the given repository.

cloudhead commented on radicle-cli/src/commands/fork.rs:
```diff
@@ -0,0 +1,74 @@
+    };
+
+    rad::fork(rid, &signer, &storage)?;
+    term::success!("forked identity for {}", profile.id());
```
```suggestion
    term::success!("Forked project {rid} for {}", profile.id());
```

cloudhead commented on radicle-cli/src/commands/fork.rs:
```diff
@@ -0,0 +1,74 @@
+    let rid = match options.rid {
+        Some(rid) => rid,
+        None => {
+            let (_, rid) = radicle::rad::repo(Path::new("."))
```
Is it even possible to have a working copy without a fork :thinking: ?

FintanH replied:
_Technically_ yes. It's possible to have fetched a project and created a git repository with the correct remote, without actually having a namespace for your NID yet. Whether that makes sense to expect is another question :smile: I did blindly copy this :see_no_evil:

cloudhead replied:
Hahaha. I guess there's no harm in leaving it like this :shrug:

FintanH commented:
I'll also add a test in `examples` for this

FintanH commented:
Made a few extra changes:
* `rad inspect` uses the canonical doc so that one can inspect a project without a fork
* Added a `rad fetch` example test
* Added the `rad fork` example test

cloudhead commented on radicle-cli/tests/commands.rs:
```diff
@@ -394,6 +394,91 @@ fn rad_init_sync_and_clone() {
+    // Necessary for now, if we don't want the new inventry announcement to be considered stale
+    // for Bob.
+    // TODO: Find a way to advance internal clocks instead.
+    thread::sleep(time::Duration::from_millis(3));
```
Do we need these sleeps in here? afaict there's only one inventory announcement.

FintanH replied:
Turns out no it's not needed :)

cloudhead commented on radicle-cli/examples/rad-fork.md:
````diff
@@ -0,0 +1,49 @@
+```
+
+We are now able to setup a remote in our own working copy of the
+project and push to our own fork.
````
Very nice
